src/fastify-plugins/maps.js

```diff
@@ -19,9 +21,39 @@ export const FALLBACK_MAP_PREFIX = 'fallback'
 /** @type {FastifyPluginAsync<MapsPluginOpts>} */
 export async function plugin(fastify, opts) {
   if (opts.customMapPath) {
+    const { customMapPath } = opts
+
+    fastify.get(`/${CUSTOM_MAP_PREFIX}/info`, async () => {
+      const baseUrl = new URL(fastify.prefix, fastify.listeningOrigin)
+
+      if (!baseUrl.href.endsWith('/')) {
+        baseUrl.href += '/'
+      }
+
+      const style = await (
+        await fetch(new URL(`${CUSTOM_MAP_PREFIX}/style.json`, baseUrl))
+      ).json()
+
+      const stats = await fs.stat(customMapPath)
+
+      const styleJsonName =
+        typeof style === 'object' &&
+        style &&
+        'name' in style &&
```

Comment:
I have this kind of check in so many places in my code and I blame typescript, I can't see any way that it's actually necessary (not saying you shouldn't have it here, just venting!)